dist/GoldRush.JS

```diff
@@ -0,0 +1,133 @@
+
+class GoldRush {
+    constructor() {
+        this.matrix = this.generateMatrix(9, 9)
+        this.player1 = {
+            score: 0
+        }
+        this.player2 = {
+            score: 0
+        }
+    }
+    get() {
+        return this.matrix
+    }
+    generateMatrix(numRows, numColumns) {
+        let newmatrix = []
+        for (let i = 0; i < numRows; i++) {
+            newmatrix[i] = []
+            for (let j = 0; j < numColumns; j++) {
+                newmatrix[i].push('block')
+            }
+        }
+        for (let c = 0; c < 70; c++) {
+            let rowCoin = Math.floor(Math.random() * 8);
+            let culCoin = Math.floor(Math.random() * 8);
+            newmatrix[rowCoin][culCoin] = 'coin'
+        }
+        for (let w = 0; w < 20; w++) {
+
+            let rowWall = Math.floor(Math.random() * 7);
+            let culWall = Math.floor(Math.random() * 7);
+            if (newmatrix[rowWall + 1][culWall] != 'wall'
+                & newmatrix[rowWall][culWall + 1] != 'wall'
+                & newmatrix[rowWall][culWall - 1] != 'wall'
+                & newmatrix[rowWall][culWall]!= 'wall') {
+                newmatrix[rowWall][culWall] = 'wall'
+            }
+        }
+        newmatrix[0][0] = 'player1'
+        newmatrix[numRows - 1][numColumns - 1] = 'player2'
+        return newmatrix
+    }
+    print() {
+        let l
+        for (let i = 0; i < 9; i++) {
+            l = ""
+            for (let j = 0; j < 9; j++) {
+                l += (this.matrix[i][j] + ' \t ')
+            }
+            console.log(l)
+        }
+    }
+    movePlayer(player, action) {
+        for (let i = 0; i < 9; i++) {
```

Comment:
make helper functions. Instead of everything being in the movePlayer method, make additional methods to break up the work. `checkWall()` `checkCoin()` would make this move player method cleaner